### PR TITLE
Random ring groups

### PR DESCRIPTION
**Pull request by badcrc**
This adds the random strategy to the ring groups, its simply does an order by random in the SQL when selecting the members of the group. It should support mysql and postgresql (though I only tested it in psql).

Also included is a lua script for entering/leaving a ring group, the code is based on the fifo_member.lua and needs a dialplan entry like in the queues.

### CODE REVIEW

File: resources/install/scripts/app/ring_groups/index.lua [hunk 1]
```diff
@@ -158,6 +158,43 @@
...
+					SELECT 
+						r.ring_group_strategy
+					FROM 
+						v_ring_groups as r, v_ring_group_destinations as d
+					WHERE 
+						d.ring_group_uuid = r.ring_group_uuid 
+						AND d.ring_group_uuid = ']]..ring_group_uuid..[[' 
+						AND r.domain_uuid = ']]..domain_uuid..[[' 
+						AND r.ring_group_enabled = 'true'  
+					]];
```
Comment (moteus) [inline resources/install/scripts/app/ring_groups/index.lua]: You do not need here join with `v_ring_group_destinations`
You can use just
```Lua
		sql = [[
				SELECT 
					r.ring_group_strategy
				FROM 
					v_ring_groups as r
				WHERE 
					AND r.ring_group_uuid = ']]..ring_group_uuid..[[' 
					AND r.domain_uuid = ']]..domain_uuid..[[' 
					AND r.ring_group_enabled = 'true'  
			]];
```

File: resources/install/scripts/app/ring_groups/index.lua [hunk 2]
```diff
@@ -158,6 +158,43 @@
 		--forward the ring group
 			session:execute("transfer", ring_group_forward_destination.." XML "..context);
 	else
+		--first we check what version of SQL are we using
+			sql = "SELECT version() as version";
```
Comment (moteus) [inline resources/install/scripts/app/ring_groups/index.lua]: `database.type` in config already has database type (e.g. `pgsql`) I think you can use it.
Checkout [directory.lua](https://github.com/fusionpbx/fusionpbx/blob/be564502df73a93edd09b0a8b11a05f49382dfcd/resources/install/scripts/app/xml_handler/resources/scripts/directory/directory.lua#L172-L177)

**Comment by badcrc** [comment]
I've updated the code with the tips from @moteus , thanks!

Also, I've included a couple of permissions fixed for issue #1065

**Comment by badcrc** [comment]
Thanks! seems that SQLite has the random() as well, MySQL seems to be the stranger one here.

**Pull request merged**